gnovm/pkg/gnolang/precompile.go

```diff
@@ -202,49 +186,31 @@ func PrecompileVerifyFile(path string, gofmtBinary string) error {
 //
 // This method is the most efficient to detect errors but requires that
 // all the import are valid and available.
-func PrecompileBuildPackage(fileOrPkg string, goBinary string) error {
+func PrecompileBuildPackage(fileOrPkg string) error {
 	// TODO: use cmd/compile instead of exec?
 	// TODO: find the nearest go.mod file, chdir in the same folder, rim prefix?
 	// TODO: temporarily create an in-memory go.mod or disable go modules for gno?
 	// TODO: ignore .go files that were not generated from gno?
 	// TODO: automatically precompile if not yet done.
 
-	files := []string{}
-
-	info, err := os.Stat(fileOrPkg)
+	files, err := gnoutil.Match([]string{fileOrPkg},
+		// no go/gno tests. disable ellipsis (must be exactly 1 package or file).
+		gnoutil.MatchFiles(`*.go`, `!/_(file)?test\.{go,gno\.gen\.go}$/`),
+		gnoutil.MatchEllipsis(false),
+		gnoutil.MatchNoImplicit(),
```

Comment:
this is possibly stretching the scope a bit (as here we're working on Go files), but given the amount of lines this removes I thought it would still be a good idea.